include/swift/AST/DiagnosticsSema.def

```diff
@@ -1015,6 +1011,21 @@ ERROR(cannot_pass_inout_arg_to_subscript,none,
       "'withUnsafeMutablePointer' to explicitly convert argument "
       "to a pointer", ())
 
+ERROR(extraneous_property_wrapper_unwrap,none,
+      "property %0 will be unwrapped to value of type %1, use '$' to refer to "
+      "wrapper type %2",
+      (DeclName, Type, Type))
+ERROR(extraneous_property_wrapper_unwrap_member_access,none,
+      "value of type %1 has no member %0, requires wrapper type %2",
+      (DeclName, Type, Type))
+ERROR(missing_property_wrapper_unwrap,none,
+      "property wrapper %0 has value of type %1, remove '$' to refer to the"
+      "wrapped value of type %2",
+      (DeclName, Type, Type))
+ERROR(missing_property_wrapper_unwrap_member_access,none,
+      "value of type %1 has no member %0, requires wrapped type %2",
+      (DeclName, Type, Type))
```

Comment:
How about we unify 2nd and last ones into `"referencing to %0 %1 requires %{wrapped value, wrapper}2 %3", (DescriptiveDeclKind, DeclName, bool, Type)` and a note to either remove or add `$`. I don't think mentioning that `$` has to be added or removed adds a lot of value to diagnostic.